src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasRelationSlicer.java

```diff
@@ -416,16 +416,26 @@ private Atlas cutSubAtlasForOriginalShard(final Atlas atlas)
 
                 // filter out all AtlasEntities that are not in the list of countries being sliced
                 // for this shard
-                final Atlas countrySubAtlas = atlas.subAtlas(isInCountry, AtlasCutType.SILK_CUT)
-                        .orElseThrow(() -> new CoreException(
-                                "Cannot have an empty atlas after relation slicing {}",
-                                getShardOrAtlasName()));
-
-                // then, filter out all remaining entities based on the shard bounds
-                return countrySubAtlas.subAtlas(this.initialShard.bounds(), AtlasCutType.SILK_CUT)
-                        .orElseThrow(() -> new CoreException(
-                                "Cannot have an empty atlas after relation slicing {}",
-                                getShardOrAtlasName()));
+                final Optional<Atlas> countrySubAtlas = atlas.subAtlas(isInCountry,
+                        AtlasCutType.SILK_CUT);
+                if (countrySubAtlas.isPresent())
+                {
+                    // then, filter out all remaining entities based on the shard bounds
+                    final Optional<Atlas> finalSubAtlas = countrySubAtlas.get()
+                            .subAtlas(this.initialShard.bounds(), AtlasCutType.SILK_CUT);
+                    if (finalSubAtlas.isPresent())
+                    {
+                        return finalSubAtlas.get();
+                    }
+                    else
```

Comment:
Is this else clause needed - can we just return null on line 438 to remove the two else clauses?

Reply:
If the first subAtlasing at L#419 was successful but the second at L#424 created an empty atlas, then without this else clause we would return the method's atlas parameter on L#442 instead of a null atlas